### PR TITLE
New Attempts table

### PR DESCRIPTION
**Pull request by joegor85**
Modified database files, rest client for testing, and server API routes. Still need to refactor front end to coordinate with these db changes.

### CODE REVIEW

File: api/src/db/migration.sql [hunk 1]
```diff
@@ -1,12 +1,17 @@
-DROP TABLE IF EXISTS tasks;
+DROP TABLE IF EXISTS attempts;
+DROP TABLE IF EXISTS question_notes;
+DROP TABLE IF EXISTS students;
+DROP TABLE IF EXISTS cohorts;
+DROP TABLE IF EXISTS questions;
+DROP TABLE IF EXISTS users;
```
Comment (jordan-carlson99) [inline api/src/db/migration.sql:6]: Joe and I couldn't figure out why these didn't exist, I'd be interested if anyone knows why

File: api/src/routesStudents.js [hunk 2]
```diff
@@ -93,43 +90,24 @@ async function remove(req, res, next) {
   }
 }
 
-async function update(req, res, next) {
```
Comment (jordan-carlson99) [inline api/src/routesStudents.js:96]: this was gross and i'll fight anyone who tries to defend it to me

File: client/src/components/InterviewCard/InterviewCard.jsx [hunk 3]
```diff
@@ -64,63 +71,111 @@ const InterviewCard = () => {
...
+    fetch(`${baseurl}/attempt`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(data),
+    }).then(() =>
+      toast("Interview Attempt saved. Redirecting...", { autoClose: 3000 })
+    );
+    fetch(`${baseurl}/student/${studentId}`, {
```
Comment (jordan-carlson99) [inline client/src/components/InterviewCard/InterviewCard.jsx:115]: the two back to back fetches aren't ideal so if anyone has some ideas on how to more elegantly do this lmk

**Review by jordan-carlson99 — COMMENTED** [review]
just leaving comments since it was 50/50 between joe and i on this one

File: api/src/db/seed.sql
Comment (joegor85) [inline api/src/db/seed.sql:1]: We made all scores to be out of 15 for simplicity and to correspond to 3 questions of 5 possible stars each.

File: api/src/routesStudents.js (hunk 2 above)
Comment (ranegray) [reply api/src/routesStudents.js:96]: i concur

File: client/src/components/InterviewCard/InterviewCard.jsx [hunk 4]
```diff
@@ -3,54 +3,61 @@ import InterviewCardCSS from "./InterviewCard.module.css";
...
+  const attempts = useUserStore((state) => state.attempts);
+  const studentName = useUserStore((state) => state.studentName);
+  const userId = parseInt(localStorage.getItem("userid"));
+  const navigate = useNavigate();
+
+  const setAttemptNumber = () => {
+    if (!studentId) {
+      console.error("StudentId not generated yet");
+      return;
+    }
```
Comment (ranegray) [inline client/src/components/InterviewCard/InterviewCard.jsx:33]: navigate back to dashboard instead here if no student is selected. remove interview link from sidebar nav.

File: client/src/components/InterviewCard/InterviewCard.jsx [hunk 5]
```diff
@@ -64,63 +71,111 @@ const InterviewCard = () => {
...
+        months[date.getMonth()]
+      }-${date.getDate()}`,
+      student_id: studentId,
+      staff_id: userId,
+      question1_id: 1,
+      rating1: rating1 || 3,
+      question2_id: 2,
+      rating2: rating2 || 3,
+      question3_id: 3,
+      rating3: rating3 || 3,
```
Comment (ranegray) [inline client/src/components/InterviewCard/InterviewCard.jsx:101]: lets default to zero.

File: client/src/components/InterviewCard/InterviewCard.jsx (hunk 3 above)
Comment (ranegray) [reply client/src/components/InterviewCard/InterviewCard.jsx:113]: might be a good use case for Promise.all and using async/await

File: client/src/components/InterviewCard/QuestionBlock.jsx [hunk 6]
```diff
@@ -1,15 +1,30 @@
...
+}]
+
+const QuestionBlock = ({ step, rating1, rating2, rating3, setRating1, setRating2, setRating3 }) => {
 
   return (
     <div className={styles.questionBlock}>
-      <p className={styles.title}>{title}</p>
-      <p>{content}</p>
-      <StarRating rating={rating} setRating={setRating} />
+      <p className={styles.title}>{questions[step-1].name}</p>
```
Comment (ranegray) [inline client/src/components/InterviewCard/QuestionBlock.jsx:23]: add comment here explaining why we are subtracting 1

File: client/src/components/StudentCard/StudentCard.jsx [hunk 7]
```diff
@@ -65,34 +46,24 @@ const StudentCard = () => {
...
       </div>
-      {/* <div className={StudentCardCSS.studentInfoCard}>
-                    <h3 className={StudentCardCSS.notesHeading}>Notes:</h3>
-                    {studentData.notes.map((note,index) => (
-                        <div key={index}>
-                            <p className={StudentCardCSS.notesInfo}>Attempt {note.attemptNum}: {note.date}</p>
-                            <p className={StudentCardCSS.notesNarrative}>{note.content}</p>
-                        </div>
-                    ))}
-                </div> */}
```
Comment (ranegray) [inline client/src/components/StudentCard/StudentCard.jsx:95]: lets keep this.

**Review by jordan-carlson99 — APPROVED** [review]

**Pull request merged**